### ISSUE TITLE
Can’t load ibec on s8001

### ISSUE THREAD

**Issue by ak99004242** [0x7ff/gaster#2]
It has an issue can’t load ibec on s8001

**Comment by 0x7ff** [comment]
The iBSS file must be signed with fake SHSH blobs so that you can load it.

**Comment by P5-2005** [comment]
same issue here, tried A10-A11, gaster pwn/patch success no issue on mac bigsur no libusb build, but cannot load ibec(signed with fake shsh blob), so tried old pwndfu with rmsigchecks same files on same devices(A10-A11) load no issue

**Comment by 0x7ff** [comment]
The latest fix may not work on S8001 (A9X) because of the cache issue I think.

**Comment by P5-2005** [comment]
> The latest fix may not work on S8001 (A9X) because of the cache issue I think.

i confirm latest fix #2 work good on A10-A11, nice

**Comment by ak99004242** [comment]
Thanks, it’s solved

**Comment by CDSQ** [comment]
iPhoneX A11 sigcheck patch can't boot ibss

**Comment by dora2-iOS** [comment]
The current commit is working correctly on A9X. I believe this project is the first to allow arbitrary unsigned iBSS on A9X! This appears to be applying a patch to a location that is not the actual ROM page...? I am very interested. when I tried to patch in the actual ROM page on the A9X, device will be reboot...

**Comment by aihkas** [comment]
For some reason, it's looping on the patch phase on s8001 using the main branch. I tried the commit after this fix with no success; the iPad either disconnected after the patching phase.
That chip is freaking tricky. Any thoughts would be appreciated.

![Screenshot](https://user-images.githubusercontent.com/1607512/194917796-8ab71527-a730-45ab-ba6f-4a932160c1d3.png)

**Comment by aihkas** [comment]
If anyone ends up in the same situation with a9x: I solved it by switching to commit `7fffffff1a62a1237b917c66a470a6bbf4612198`.
And make sure to copy the executable to `/usr/local/bin/` after making the file.

**Comment by aihkas** [comment]
> It has an issue can’t load ibec on s8001

Did you manage to get firmware keys for s8001?

**Comment by leocg** [comment]
Did you manage to send iBEC to an A9X device?

Device enter pwned dfu using latest commit with gaster pwn but i'm getting stuck sending iBEC.

<img width="610" alt="Screenshot 2023-03-02 at 12 02 10" src="https://user-images.githubusercontent.com/3638669/222470036-725c8491-11ab-441e-b4d8-db160a4872f6.png">

Also tried @dora2-iOS commit of ipwndfu supporting A9X, but get "Device not supported for --repair-heap" and iPad reboot trying ipwndfu --patch-sigchecks

**Comment by aihkas** [comment]
> Also tried @dora2-iOS commit of ipwndfu supporting A9X, but get "Device not supported for --repair-heap" and iPad reboot trying ipwndfu --patch-sigchecks

dora2-iOS's branch on ipwndfu won't work. I managed to get past that step and successfully sent iBEC, as I mentioned above, with gaster. I also managed to extract the firmware keys, but I had another problem that required a lot of work, and by that time, palera1n was released, which achieved my purpose and helped me experiment with this iPad. What are you trying to achieve here, if I may ask, in order to help you?
